File: Editorial/165_Compare_Version_Numbers/Go/1_Split_Parse_Two_Pass.go
func compareVersion(version1 string, version2 string) int {
    tokens1 := strings.Split(version1, ".")
    tokens2 := strings.Split(version2, ".")

    for i := 0; i < max(len(tokens1), len(tokens2)); i++ {
        i1 := 0
        i2 := 0
        if i < len(tokens1) {
            i1, _ = strconv.Atoi(tokens1[i])
        }
        if i < len(tokens2) {
            i2, _ = strconv.Atoi(tokens2[i])
        }

        if i1 != i2 {
            if i1 > i2 {
                return 1
            }
            return -1
        }
    }

    // The versions are equal
    return 0
}

File: Editorial/165_Compare_Version_Numbers/C/1_Split_Parse_Two_Pass.c
int compareVersion(char* version1, char* version2) {
    while (*version1 || *version2) {
        long num1 = strtol(version1, &version1, 10);
        long num2 = strtol(version2, &version2, 10);

        if (num1 > num2) return 1;
        if (num1 < num2) return -1;

        if (*version1) version1++;
        if (*version2) version2++;
    }

    return 0;
}
